core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java

```diff
@@ -742,6 +751,11 @@ public void dropColumn(Session session, TableHandle tableHandle, ColumnHandle co
         CatalogHandle catalogHandle = tableHandle.getCatalogHandle();
         ConnectorMetadata metadata = getMetadataForWrite(session, catalogHandle);
         metadata.dropColumn(session.toConnectorSession(catalogHandle), tableHandle.getConnectorHandle(), column);
+
+        CatalogMetadata catalogMetadata = getCatalogMetadataForWrite(session, catalogHandle);
+        ColumnMetadata sourceColmnMetadata = getColumnMetadata(session, tableHandle, column);
```

Comment:
I don't think this works after the column is dropped. Also, we should probably do this conditionally, only when using system security.

Reply:
> I don't think this works after the column is dropped.

Changed to capture the source column name before dropping the column.

> Also, we should probably do this conditionally, only when using system security.

Changed all three column event methods to only send the events if system access control is in effect.